parser_test.go

```diff
@@ -20,25 +20,25 @@ var messageTests = []struct {
 	// Total output
 	Expect   string
 	ExpectIn []string
-	IsNil    bool
+	Err      string
 
 	// FromChannel
 	FromChan bool
 }{
 	{
-		IsNil: true,
+		Err: "Cannot parse zero-length message",
```

Comment:
The above suggestion also makes this easier to parse, you can compare the Err by changing it to type `error` instead of having to do a string comparison.